### PR TITLE
OPENEUROPA-1523: Create the EWCMS site template.

### PR DESCRIPTION
**Pull request by sergepavle**

### CODE REVIEW

File: scripts/composer/SetupWizard.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    $sanitized_organization_name = preg_replace('/[^a-zA-Z ]/', '', trim($organization_name));
+    $camelcased_organization_name = preg_replace('/ /', '', ucwords($sanitized_organization_name));
+
+    $machine_name = preg_replace('/\s+/', '_', $sanitized_project_name);
+    $machine_name = $event->getIO()->ask('<info>What is the (machine readable) project name?</info> [<comment>' . $machine_name . '</comment>]? ', $machine_name);
+
+    $package_name = preg_replace('/\s+/', '-', strtolower($sanitized_organization_name)) . '/' . preg_replace('/\s+/', '-', $sanitized_project_name);
+    $package_name = $event->getIO()->ask('<info>What is the package name?</info> [<comment>' . $package_name . '</comment>]? ', $package_name);
+
+    $description = 'A website built on the OpenEuropa platform.';
```
Comment (voidtek) [inline scripts/composer/SetupWizard.php]: We don't need to specify that is an EWCMS project, here?

File: src/Context/FeatureContext.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Defines step definitions that are generally useful for the project.
+ */
+class FeatureContext extends RawDrupalContext {
+
+  /**
+   * Checks that a 200 OK response occurred.
+   *
+   * @Then I should get a valid web page
```
Comment (voidtek) [inline src/Context/FeatureContext.php]: This step is missing to put on the behat tests.

File: config/sync/README.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+This directory contains configuration to be imported into your Drupal site. To
```
Comment (voidtek) [inline config/sync/README.txt]: We should remove this file and put a `.gitkeep` file.

File: grumphp.yml.dist [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    - web/
+  tasks.phpcs.triggered_by:
+    - php
+    - module
+    - inc
+    - theme
+    - install
+
+  extra_tasks:
+    phpparser:
```
[the remark this answers is not in the document]
Comment (drupol) [reply grumphp.yml.dist]: I just stumbled accross this in my emails... we can get rid of this extra task, it won't be necessary in the next version of `openeuropa/code-review`.

File: behat.yml.dist [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      javascript_session: 'selenium2'
+      selenium2:
+        wd_host: '${selenium.host}:${selenium.port}/wd/hub'
+        capabilities: { "browser": "chrome", "version": "*" }
+      base_url: "${drupal.base_url}"
+    Drupal\DrupalExtension:
+      api_driver: "drupal"
+      drupal:
+        drupal_root: "${drupal.root}"
+      region_map:
```
[the remark this answers is not in the document]
Comment (ademarco) [reply behat.yml.dist]: We should still use Drupal extension region map, the transformation context will not hook into those.

File: src/Context/FeatureContext.php (hunk 2 above)
Comment (voidtek) [reply src/Context/FeatureContext.php]: This comment is not resolved.

File: .drone.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    image: fpfis/httpd-php-dev:7.1
+    environment:
+      - DOCUMENT_ROOT=/test/ewcms-site
+      - COMPOSER_CACHE_DIR=/cache/composer
+  mysql:
+    image: percona/percona-server:5.6
+    environment:
+      - MYSQL_ALLOW_EMPTY_PASSWORD=yes
+  selenium:
+    image: selenium/standalone-chrome:3.11
```
Comment (voidtek) [inline .drone.yml]: We should use the `standalone-chrome:3.14`

File: .drone.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
+workspace:
+  base: /test
+  path: ewcms-site
```
Comment (voidtek) [inline .drone.yml]: We should use:
```
  base: /var/www
  path: html
```

File: .drone.yml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+services:
+  web:
+    image: fpfis/httpd-php-dev:7.1
+    environment:
+      - DOCUMENT_ROOT=/test/ewcms-site
```
Comment (voidtek) [inline .drone.yml]: According to the previous message, we should change this line.

File: .gitignore [hunk 9]
```diff
@@ -0,0 +1,50 @@
+# Ignore Composer dependencies.
```
Comment (voidtek) [inline .gitignore]: This file is not aligned with `drupal-site-template`. Why? 
And it seems the same as the `.dist`. Why?

File: README.md [hunk 10]
```diff
@@ -1 +1,69 @@
...
+set of Web Content Management (WCM) features, enables building websites in a few clicks and is easily extensible.
+
+This project template provides a starter kit for creating a website using the
+OpenEuropa Drupal 8 platform. It will install the [EWCMS Profile](https://github.com/openeuropa/ewcms)
+which is an installation profile that includes the OpenEuropa components
+ and the [OpenEuropa Theme](https://github.com/openeuropa/oe_theme).
+Using this theme will ensure that the project complies with the guidelines for
+[European Component Library](https://github.com/ec-europa/europa-component-library).
+
+## Prerequisites
```
Comment (voidtek) [inline README.md]: This paragraph doesn't make sense with this `README.md`

File: README.md.dist [hunk 11]
```diff
@@ -0,0 +1,239 @@
+# OpenEuropa template for EWCMS projects
+
+This is based on [drupal-composer/drupal-project](https://github.com/drupal-composer/drupal-project).
+
+[![Build Status](https://drone.fpfis.eu/api/badges/openeuropa/ewcms-site-template/status.svg?branch=master)](https://drone.fpfis.eu/openeuropa/ewcms-site-template)
+
+The Europa Web Content Management System (EWCMS) is a state of the art web publishing platform for building EU and EC
```
Comment (voidtek) [inline README.md.dist]: This part is not coherent in this `README.md`

File: README.md.dist [hunk 12]
```diff
@@ -0,0 +1,239 @@
+# OpenEuropa template for EWCMS projects
```
Comment (voidtek) [inline README.md.dist]: The title of this page should be dynamic.

File: README.md.dist [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+[European Component Library](https://github.com/ec-europa/europa-component-library).
+
+## Prerequisites
+
+You need to have the following software installed on your local development environment:
+
+* [Composer](https://getcomposer.org/doc/00-intro.md#installation-linux-unix-osx).
+* [Docker Compose](https://docs.docker.com/compose/install/)
+
+## Create the project
```
Comment (voidtek) [inline README.md.dist]: This paragraph is for EWCMS site template, not for the final site.

File: README.md.dist [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+A `.drone.yml` file is provided for running CI tests on Drone. Further details of how to set this up can be found in the
+ [drone documentation](https://docs.drone.io/).
+
+## Project management
+
+It is recommended that the version of `oe_theme` is locked to the current minor version before going live with the
+project, so that updates to the theme do not cause problems to a running site. We recommend that this is periodically
+updated to the latest version, after doing manual testing.
+
+A separate `.gitignore` file is provided which is used for the project. Drupal scaffold files should be committed after
```
Comment (voidtek) [inline README.md.dist]: This paragraph should not be here.

File: README.md.dist [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+Starting a new project is a 4 step procedure:
+
+## 0. Prerequisites
+
+You need to have the following software installed on your local development environment:
+
+* [Composer](https://getcomposer.org/doc/00-intro.md#installation-linux-unix-osx).
+* [Docker Compose](https://docs.docker.com/compose/install/)
+
+## 1. Create the project
```
Comment (voidtek) [inline README.md.dist]: This paragraph is a duplicate.

File: behat.yml.dist [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+  extensions:
+    Drupal\MinkExtension:
+      goutte: ~
+      browser_name: '${selenium.browser}'
+      javascript_session: 'selenium2'
+      selenium2:
+        wd_host: '${selenium.host}:${selenium.port}/wd/hub'
+        capabilities: { "browser": '${selenium.browser}', "version": "*" }
+      base_url: "${drupal.base_url}"
+    Drupal\DrupalExtension:
```
Comment (voidtek) [inline behat.yml.dist]: See how add the `ajax_timeout` parameter.

File: composer.json [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+        "easyrdf/easyrdf": "0.10.0-alpha.1 as 0.9.1",
+        "openeuropa/ewcms": "dev-master",
+        "openeuropa/oe_theme": "^0.9",
+        "webflo/drupal-finder": "^1.0",
+        "webmozart/path-util": "^2.3"
+    },
+    "require-dev": {
+        "drupal/console": "~1.6",
+        "drupal/drupal-extension": "~4.0",
+        "openeuropa/behat-transformation-context": "^0.1.1",
```
Comment (voidtek) [inline composer.json]: should be `"openeuropa/behat-transformation-context": "~0.1",`

File: composer.json [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+        "openeuropa/code-review": "^1.0.0-beta1",
```
Comment (voidtek) [inline composer.json]: Should be `"openeuropa/code-review": "~1.0",`

File: features/authentication.feature [hunk 19]
```diff
@@ -0,0 +1,72 @@
+Feature: User authentication
+  In order to protect the integrity of the website
+  As a product owner
+  I want to make sure users with various roles can only access pages they are authorized to
+
+  Scenario: Anonymous user can see the user login page
+    Given I am not logged in
+    When I visit "user"
```
[the remark this answers is not in the document]
Comment (voidtek) [reply features/authentication.feature]: Remark not resolved.

File: features/authentication.feature [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+    And I should see the text "Reset your password"
+    And I should see the text "Username"
+    And I should see the text "Password"
+    But I should not see the text "Log out"
+    And I should not see the text "View profile"
+
+  Scenario Outline: Anonymous user can access public pages
+    Given I am not logged in
+    # By visiting the path an access check is done automatically.
+    Then I visit "<path>"
```
[the remark this answers is not in the document]
Comment (voidtek) [reply features/authentication.feature]: Remark not resolved.

File: features/authentication.feature [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+
```
[the remark this answers is not in the document]
Comment (voidtek) [reply features/authentication.feature]: Remark not resolved.

File: features/authentication.feature [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+    Examples:
```
[the remark this answers is not in the document]
Comment (voidtek) [reply features/authentication.feature]: Remark not resolved.

File: features/authentication.feature [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+      | user/login    |
+      | user/password |
+      | user/register |
+
+  Scenario Outline: Anonymous user cannot access restricted pages
+    Given I am not logged in
+    When I go to "<path>"
+    Then I should get an access denied error
+
+    Examples:
```
[the remark this answers is not in the document]
Comment (voidtek) [reply features/authentication.feature]: Remark not resolved.

File: features/authentication.feature [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+      | path             |
+      | admin            |
+      | admin/appearance |
+      | admin/config     |
+      | admin/content    |
+      | admin/people     |
+      | admin/structure  |
+      # To do: the 'node/' path should be disabled on most Drupal sites.
```
[the remark this answers is not in the document]
Comment (voidtek) [reply features/authentication.feature]: Remark not resolved.

File: features/authentication.feature [hunk 25]
```diff
@@ -0,0 +1,72 @@
...
+      # | node            |
+      | node/add         |
+      | user/1           |
+
+  @api
+  Scenario Outline: Authenticated user can access pages they are authorized to
+    Given I am logged in as a user with the "authenticated" role
+    Then I visit "<path>"
```
[the remark this answers is not in the document]
Comment (voidtek) [reply features/authentication.feature]: Remark not resolved.

File: src/Context/FeatureContext.php [hunk 26]
```diff
@@ -0,0 +1,38 @@
...
+   *   Thrown when a different HTTP response code was returned.
+   */
+  public function assertAccessDenied(): void {
+    $this->assertSession()->statusCodeEquals(403);
+  }
+
+  /**
+   * Checks that a 200 OK response occurred.
+   *
+   * @Then I should get a valid web page
```
Comment (voidtek) [inline src/Context/FeatureContext.php]: This step is not used in this PR.

**Comment by voidtek** [comment]
I push some fix to run correctly the test on Drone. I will create issues on Github to resolve the following tasks.

**Pull request merged**